lua/toggleterm/terminal.lua

```diff
@@ -256,6 +256,7 @@ function Terminal:shutdown()
     self:close()
   end
   ui.delete_buf(self)
+  delete(self.id)
```

Comment:
This should not affect anything since shutdown is not called by anything but the test cases which should pass.